src/clojure_lsp/handlers.clj

```diff
@@ -136,29 +136,58 @@
     (async/pipeline-blocking 5 output-chan xf (async/to-chan dirs) true (fn [e] (log/warn e "hello")))
     (async/<!! (async/into {} output-chan))))
 
-(defn lookup-classpath [project-root]
+(defn is-lein-project? [project-root]
+  (.exists (io/file project-root "project.clj")))
+
+(defn is-boot-project? [project-root]
+  (.exists (io/file project-root "build.boot")))
+
+(defn hash-project-file [project-file]
+  (digest/md5 project-file))
+
+(defn lookup-classpath [command root-path]
   (try
-    (let [root-path (uri->path project-root)
-          sep (re-pattern (System/getProperty "path.separator"))]
-      (-> (shell/sh "lein" "classpath" :dir root-path)
+    (let [sep (re-pattern (System/getProperty "path.separator"))]
+      (-> (command)
           (:out)
           (string/trim-newline)
           (string/split sep)))
     (catch Exception e
-      (log/warn "Could not run lein in" project-root (.getMessage e)))))
+      (log/warn "Error while looking up classpath info in" root-path (.getMessage e))
+      '[])))
+
+(defn lein-lookup-classpath [root-path]
+  (lookup-classpath #(shell/sh "lein" "classpath" :dir root-path) root-path))
+
+(defn boot-lookup-classpath [root-path]
+  (lookup-classpath #(shell/sh "boot" "show" "--fake-classpath" :dir root-path) root-path))
+
+(defn build-lein-project [root-path]
+  (let [file-hash (hash-project-file (io/file root-path "project.clj"))
+        classpath (lein-lookup-classpath root-path)]
+    {:project-hash file-hash :classpath classpath}))
+
+(defn build-boot-project [root-path]
+  (let [file-hash (hash-project-file (io/file root-path "build.boot"))
+        classpath (lein-lookup-classpath root-path)]
```

Comment:
Should be `boot-lookup-classpath`

Reply:
Thanks, that was an obvious one. I think the issue was that my tests solely consisted of seeing if I get some autocompletion for a boot project (which isn't enought, as this is done by nrepl I think).